OTAcceleratorCore/OTMultiPartyCommunicator.m

```diff
@@ -385,10 +394,27 @@ - (void)session:(OTSession *)session streamDestroyed:(OTStream *)stream {
     }
 }
 
+- (void)session:(OTSession*) session
+connectionCreated:(OTConnection*) connection {
+    _internalConnectionCount++;
+    
+    //check creationtime of the connections
+    [self compareConnectionTimeWithConnection: connection];
+}
+
+- (void)session:(OTSession*) session
+connectionDestroyed:(OTConnection*) connection {
+    _internalConnectionCount--;
+    
+    //check creationtime of the connections
+    [self compareConnectionTimeWithConnection: connection];
+}
```

Comment:
For calculating the `connectionCount`, we can take advantage of the active streams by the following implementation so `connectionCreated:` and `connectionDestroyed:` are not needed to reduce the code complexity. 
```
func connectionCount() -> Int {
        var streams = [String: Int]()
        for stream: OTStream in AppDelegate.getSharedAcceleratorSession().streams.values {
            if streams[stream.connection.connectionId] != nil {
                streams[stream.connection.connectionId]! += 1
            }
            else {
                streams[stream.connection.connectionId] = 1
            }
        }
        return streams.count
    }
```